### PR TITLE
doc_review should return failure on malformed rns

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/48051)

### CODE REVIEW

**Comment by DinaMeylakh** [comment]
Failing [test](https://app.circleci.com/pipelines/github/demisto/demisto-sdk/15895/workflows/90950ebc-9952-46fa-85fd-f1aeca029267/jobs/95251) in [3d5de2b](https://github.com/demisto/demisto-sdk/pull/2038/commits/3d5de2b27aa4023534012f6ff2a68d48e49434d9)

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,7 +33,6 @@ the same as the last release note version.
 * Fixed an issue in the **validate** and **format** commands where *None* arguments in integration commands caused the commands to fail unexpectedly.
 * Added support for running tests on XSIAM machines in the **test-content** command.
 * Fixed an issue where the **validate** command did not work properly when deleting non-content items.
-* Added the flag '-d', '--dependency' to **find-dependencies** command to get the content items that cause the dependencies between two packs.
```
Comment (GuyAfik) [inline CHANGELOG.md:34]: why did you remove it?

File: demisto_sdk/commands/doc_reviewer/tests/doc_reviewer_test.py [hunk 2]
```diff
@@ -268,6 +268,29 @@ def test_valid_spelled_files(self, valid_spelled_content_pack):
...
+        """
+        rn = pack.create_release_notes(
+            version="release-note-0",
+            content="\n#### Script\n##### Script Name\n- blah blah"
+        )
+        doc_reviewer = DocReviewer(file_paths=[pack.path])
+        result = doc_reviewer.run_doc_review()
+        assert rn.path in doc_reviewer.malformed_rn_files
+        assert not doc_reviewer.found_misspelled
+        assert result is False
```
Comment (GuyAfik) [inline demisto_sdk/commands/doc_reviewer/tests/doc_reviewer_test.py:292]:
```suggestion
        assert not result
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Changelog
 
+demisto_sdk.tests.integration_tests.content_create_artifacts_integration_test
```
Comment (GuyAfik) [inline CHANGELOG.md:3]:
```suggestion
Fixed an issue where the **doc-review** command was successful even when the release-note is malformed.
```
not sure what is your current change log here?

**Review by GuyAfik — APPROVED** [review]
@DinaMeylakh looks good, nice work!

See some minor comments.

File: CHANGELOG.md (hunk 3 above)
Comment (DinaMeylakh) [reply CHANGELOG.md]: bad merge, sorry
